### PR TITLE
feat: Add Ruby support for auto code mappings

### PR DESCRIPTION
**Pull request by jangjodi**
Fixes WOR-2467

To the best of my knowledge, Ruby is already supported for auto code mappings. This PR just adds tests to confirm this.

This [query](https://sentry.io/organizations/sentry/discover/homepage/?cursor=0%3A0%3A1&field=stacktrace_link.file&field=count%28%29&field=http.referer&field=stacktrace_link.in_app&name=All+Events&project=1&query=transaction%3A%2Fapi%2F0%2Fprojects%2F%7Borganization_slug%7D%2F%7Bproject_slug%7D%2Fstacktrace-link%2F+stacktrace_link.platform%3Aruby&sort=-stacktrace_link.file&statsPeriod=14d&yAxis=count%28%29) shows Ruby stack trace files. They all are straight paths, and can use the [_straight_path_logic function](https://github.com/getsentry/sentry/blob/49f4235cacaa5ef61de7bf05b7a2d8ee6a8aab38/src/sentry/integrations/utils/code_mapping.py#L115) to determine the auto code mapping.

### CODE REVIEW

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 1]
```diff
@@ -185,6 +185,60 @@ def test_derive_code_mappings_one_to_one_match(self):
...
+            ]
+        )
+
+    def test_find_stacktrace_paths_single_project(self):
+        stacktrace_paths = identify_stacktrace_paths(self.event_data)
+        assert set(stacktrace_paths) == {"some/path/test.rb", "lib/tasks/crontask.rake"}
+
+    @responses.activate
+    @with_feature("organizations:derive-code-mappings")
+    def test_derive_code_mappings_rb(self):
```
Comment (jangjodi) [inline tests/sentry/tasks/test_derive_code_mappings.py:211]: Tests derive code mapping for `some/path/test.rb`

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 2]
```diff
@@ -185,6 +185,60 @@ def test_derive_code_mappings_one_to_one_match(self):
...
+            }
+            derive_code_mappings(self.project.id, self.event_data)
+            code_mapping = RepositoryProjectPathConfig.objects.all()[0]
+            assert code_mapping.stack_root == ""
+            assert code_mapping.source_root == ""
+            assert code_mapping.repository.name == repo_name
+
+    @responses.activate
+    @with_feature("organizations:derive-code-mappings")
+    def test_derive_code_mappings_rake(self):
```
Comment (jangjodi) [inline tests/sentry/tasks/test_derive_code_mappings.py:227]: Tests derive code mapping for `lib/tasks/crontask.rake`

**Review by snigdhas — APPROVED** [review]

**Review by armenzg — CHANGES_REQUESTED** [review]
It won't start happening until it gets added here:
https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/post_process.py#L610-L612

**Comment by jangjodi** [comment]
> It won't start happening until it gets added here: https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/post_process.py#L610-L612

Good catch, thank you

**Review by armenzg — APPROVED** [review]
Feel free to merge it.

**Pull request merged**